### PR TITLE
Adds rest api for cluster shutdown. Updated stop.sh script

### PR DESCRIPTION
**Pull request by kobalski**

### CODE REVIEW

File: hazelcast/src/main/resources/stop.sh [hunk 1]
```diff
@@ -1,13 +1,104 @@
...
+
+if [[ -z "$SCOPE" ]]; then
+ 	echo "No scope is defined, running script with default scope : 'member'"
+ 	SCOPE="member"
+fi
+	
+if [ $SCOPE = "cluster" ]; then
+	
+	if [[ -z "$PORT" ]]; then
+		echo "No port is defined running script with default port : '5701'"
```
Comment (emrahkocaman) [inline hazelcast/src/main/resources/stop.sh]: "," or "." needed after `No port is defined`. (Same thing for below messages)

File: hazelcast/src/test/java/com/hazelcast/internal/ascii/RestTest.java [hunk 2]
```diff
@@ -182,4 +184,28 @@ public void testDisabledRest() throws IOException {
...
+                assertFalse(instance2.getLifecycleService().isRunning());
+            }
+        });
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run()
+                    throws Exception {
+                assertFalse(instance2.getLifecycleService().isRunning());
+            }
+        });
```
Comment (emrahkocaman) [inline hazelcast/src/test/java/com/hazelcast/internal/ascii/RestTest.java]: Looks like both `assertTrueEventually` checks the same condition.

File: hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpPostCommandProcessor.java [hunk 3]
```diff
@@ -44,6 +48,9 @@ public void handle(HttpPostCommand command) {
                 handleManagementCenterUrlChange(command);
             } else if (uri.startsWith(URI_QUEUES)) {
                 handleQueue(command, uri);
+            } else if (uri.startsWith(URI_SHUTDOWN_CLUSTER_URL)) {
+                handleClusterShutdown(command);
+                return;
```
Comment (emrahkocaman) [inline hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpPostCommandProcessor.java]: Is `return` necessary here? Can't we just set the response of `command` and let the response be sent in line 60.

Comment (emrahkocaman) [inline hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpPostCommandProcessor.java]: ah ok. we need to send response before shutdown.

**Comment by noctarius** [comment]
Is it really a good idea to use groupname and grouppassword as a security feature, in general those two properties are not considered to be used for security? In general is it really a good idea to provide shutdown on the same port as typical rest api calls?

**Comment by mesutcelik** [comment]
I believe user/pass is used for pre-caution rather than security feature. We can even add one more warning like `You are shutting down the cluster. please provide user/pass?`. @kobalski ?

Rest API is already used by management center to change the management center URL so this is another addition.

File: hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpPostCommandProcessor.java (hunk 3 above)
Comment (kobalski) [inline hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpPostCommandProcessor.java]: Yes, exactly.

File: hazelcast/src/test/java/com/hazelcast/internal/ascii/RestTest.java (hunk 2 above)
Comment (kobalski) [inline hazelcast/src/test/java/com/hazelcast/internal/ascii/RestTest.java]: Done. Also did not need two separate `assertTrueEventually `.

File: hazelcast/src/main/resources/stop.sh (hunk 1 above)
Comment (kobalski) [inline hazelcast/src/main/resources/stop.sh]: Added ','.

**Comment by kobalski** [comment]
@noctarius @mesutcelik groupname and pass is only used for precaution. This PR is intended to provide cluster shutdown with our `stop.sh` script for Hot Restart feature.  In 3.7, we should have more secure and more well designed Cluster Management API and discuss all these concerns during the design of that API.

**Comment by kobalski** [comment]
@bilalyasar I renamed `server.sh` to `start.sh`, Do we need to update our release script after this change.

**Comment by mesutcelik** [comment]
renaming might not be a good option. It was always `server.sh` and people are used to this naming...

**Comment by kobalski** [comment]
@mesutcelik I reverted renaming of `server.sh`

**Comment by mesutcelik** [comment]
:+1:

**Comment by emrahkocaman** [comment]
:+1:

File: hazelcast/src/main/resources/stop.sh [hunk 4]
```diff
@@ -1,13 +1,105 @@
...
+
+	if [[ -z "$PASSWORD" ]]; then
+		echo "No password is defined, running script with default password : 'dev-pass'"
+		PASSWORD="dev-pass"
+	fi
+	echo "You are shutting down the cluster. Please make sure you provide valid user/pass."
+	echo "Shutting down cluster from member on this machine on port ${PORT}"
+
+	request="http://127.0.0.1:${PORT}/hazelcast/rest/management/cluster/shutdown"
+ 	response=$(curl --data "${GROUPNAME}&${PASSWORD}" --silent "${request}");
```
Comment (mdogan) [inline hazelcast/src/main/resources/stop.sh]: Shouldn't you check if `curl` is available and print a warning if not?

Comment (kobalski) [inline hazelcast/src/main/resources/stop.sh]: Good idea, I added check here : https://github.com/hazelcast/hazelcast/pull/6929/files#diff-d8a0ca814fb592a4a2d706bc7d79e76dR48

**Comment by mdogan** [comment]
:+1:

**Pull request merged**